### DISCUSSION TITLE
Reinstallation question

### DISCUSSION THREAD

**Discussion by ldb20151** [aaronwmorris/indi-allsky#1726] [Q&A]
After struggling through my GPS and time sync issues, I had the camera working great, albeit not finished.  I sat down to install a pile of pi os updates that had been nagging me the past couple of weeks and in the middle of the installation or power failed.  I don't know where it was in the update but the result was that the pi was essentially bricked by the power glitch.  Bad on me for not using an UPS and keeping a drive image.  So....I proceeded to reinstall the pi os and indi-allsky per the wiki.  I'm using a pi 5 (8mb) with the adafruit HQ imx477 camera and ultimate GPS hat.  Anyway, as I walked through the allsky installation prompts, I selected the for the onboard camera interface and the imx477 interface which is what the HQ camera spec calls out.  At this point, Indi-allsky forced a custom build as no pre-built files were available for my configuration -- which seemed odd and different than the last time I ran this installation script back in Oct.  It took forever to complete the installation, and once done, it took forever for the camera to show up as available in the webpage (about 45min).  

Everything "seems" to be working ok, although I have not yet completed the GPS and time sync steps that previously gave me fits.  I'm expecting those will be easier this time around.  I want to be sure the installation is OK and isn't likely to become unstable in the future because I somehow missed something during this second installation.  If there's doubt, I'd rather stop here and redo it again than find issues down stream.  Once I get the GPS connected I plan to make a clone drive so I don't need to revert to total installation if something happens again.  The bottom line, should I be worried that the script forced me into a custom build given my hw config?

v/r,
Larry

**Comment by aaronwmorris** [comment]
The time it takes to compile the INDI code sounds a bit high.  If you do not have a fan/heatsink on your system, the heat can cause the system to throttle which can slow down the compile significantly.  Since you are using libcamera, you only need the indi-core to be built which can cut the compile time in half, but it does not hurt to have the core and 3rd-party stuff built.

Having to wait so long for images, is a bit unusual, but this can sometimes happen when the timezone is changed on the first build which affects the timestamps and can cause things to lag a bit.

If it is working now, I expect it to continue operating normally.

**Reply by ldb20151** [reply]
I do have detailed build notes from the first time around and I see that my first install was done with the pi free-standing as the case I ultimately found to fit my NVME bottom hat configuration had not yet arrived.  I do have active cooling but as I look at the new case, which was installed for the second build, I expect cooling is very poor; especially for the SSD.  I'm re-thinking the pi enclosure and whole cooling process.  Your speculation that heating affected the re-build time is likely correct.  

The system when connected locally does seem to work correctly with one exception.  I'm unable to access the indi-allsky web from another PC on my LAN via https://<IP Address>/indi-allsky/  This worked perfectly the last install with no effort on my part except that at the time of the power failure, it would only connect via http://  vs https:// .  I never had a chance to explore that issue in the first build.  Since I cannot connect remotely at all this build, its a bigger issue.  Any thoughts on what would cause this?  I'm still looking through the wiki but haven't found anything to explain it.

Otherwise, the only thing I notice is whenever I do a cold start, it takes longer I think for the initial image to appear than in the first build -- more than 15 minutes.  Once it shows up, the image updates approximately every 20 sec (which seems normal to me).  GPS and NTP has been reaccomplished with no issues.
v/r,
-Larry

**Reply by ldb20151** [reply]
Update:  This system modified my LAN address above;  to avoid confusion, I'm trying to access the camera at: https:// my pi's IP address/indi-allsky/  This does not work.  System response is 'The Site Cannot Be Reached'

**Reply by aaronwmorris** [reply]
There some other configurations that have to be in place to allow a URL like https://indi-allsky/ to work on your home network.  It would probably be better to use something like https://hostname.local/ (the .local is important) which would use the mDNS (Bonjour) resolution.  Some home networks use .lan as the extension which might work.  There are a lot of variables that cannot be controlled by indi-allsky.

I would expect it would take about 3-5 minutes for the first image to appear after a cold start.  The indi-allsky service is started by timer 2 minutes after boot and it could take another 1-2 minutes for the initialization to complete.  If the system is using GPS exclusively for time (and not NTP) there could be some waiting for GPS to get a lock.  When a raspberry pi is powered off, it saves the last time and that time is restored when booted, so if the system is powered off for an hour, the time will be an hour behind until a time sync can occur.

**Reply by ldb20151** [reply]
OK.  That makes sense and is consistent with what I'm seeing.  At this point I've powered the system on/off several times from 'cold' start and the time it takes to load is consistent to within about 30 seconds -- that could be GPS driven time requirements.
I have SSH back in place for remote maintenance (I plan to put this on an inconvenient position on a pole) and remote web access looks good too, albeit it ONLY worked via http://.  I tried the https://hostname.local/indi-allsky as suggested.  It connects just fine but still refuses https connection.  I've tried multiple browsers as well (Chrome and Firefox) without luck.  Seems like I must have something misconfigured but I'm not sure where to look -- browser side or in the pi?  

As for my main connection issue (i.e. unable to connect to pi remotely at all) I don't know what I did to fix it but its working and appears solid as it has continued to perform flawlessly since first connection today -- thru multiple reboots and multiple browsers.  I was walking through router settings, checking the firewall and then configured SSH with associated key pair.  When I rebooted to check SSH with Mobaxterm I noticed the PC was connected to the web page (http only).  So, I obviously had something misconfigured.  Likely the firewall.  I have those settings locked down and (hopefully) saved to a config file.

Anyway, I don't plan today to use this via internet outside my LAN so can use it as is, but would feel better from security perspective if I could get an https connection.  I'm willing to do the work if someone can point me in the right direction to fix this.  Looks like indi-allsky uses nginx as reverse proxy to handle self signed certificates?  I'm assuming anything required to make that work was handled at setup? As a professional Aerospace Engineer, I know enough not to try to mess with this part of the code.  Its WAY over my head!
